infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServerCache.java

```diff
@@ -45,7 +46,10 @@ public LogUnitServerCache(LogUnitServerConfig config, StreamLog streamLog) {
      */
     private ILogData handleRetrieval(long address) {
         LogData entry = streamLog.read(address);
-        log.trace("handleRetrieval: Retrieved[{} : {}]", address, entry);
+        log.debug("handleRetrieval: Retrieved[{} : {}]", address, entry);
+        if (entry != null && !(streamLog instanceof InMemoryStreamLog)) {
+            entry.serializeMetadata();
```

Comment:
the checking of log type is memory should be removed. As many of our test cases are tested against memory log.